Add VPN Moments page - October

nathan-barrett opened the pull request:
## One-line summary

Add VPN moments page - October (Fixes #12233)

## Significant changes and points to review
This page uses the same layout and shares fluent strings with the moments page-10

## Issue / Bugzilla link
#12233 

## Checklist
- [ ] Coupon code hidden behind switch


## Testing

Demo server URL: http://localhost:8000/firefox/welcome/14/

alexgibson commented on bedrock/firefox/templates/firefox/welcome/page14.html:
```diff
@@ -0,0 +1,82 @@
+        </div>
+      {% endif %}
+      <div class="c-main-cta">
+        <a href="{{ url('products.vpn.landing') }}{{ source }}#pricing" class="mzp-c-button mzp-t-product mzp-t-xl" data-cta-text="Get Mozilla VPN" data-cta-type="button">{{ ftl('welcome-page10-get-mozilla-vpn') }}</a>
```
Issue (blocking): This should use `vpn_product_referral_link()` (instead of `{{ source }}`) to track subscription referrals: https://bedrock.readthedocs.io/en/latest/firefox-accounts.html#tracking-same-site-links-for-mozilla-vpn

nathan-barrett replied:
oh, duh! I just grabbed the contents of `page10` since it's the same design

alexgibson commented on bedrock/firefox/urls.py:
```diff
@@ -129,6 +129,7 @@
     page("firefox/welcome/11/", "firefox/welcome/page11.html", ftl_files=["firefox/welcome/page11"]),
     page("firefox/welcome/12/", "firefox/welcome/page12.html", active_locales=["en-US", "en-CA", "en-GB"]),
     page("firefox/welcome/13/", "firefox/welcome/page13.html", ftl_files=["firefox/welcome/page13"]),
+    page("firefox/welcome/14/", "firefox/welcome/page14.html", ftl_files=["firefox/welcome/page10", "firefox/welcome/page14"]),
```
Issue (maybe blocking?): it's a little unusual to be mixing two different FTL files like this. The first file in the array will be used for activation, meaning all locales that have `page10.ftl` translated will be enabled in production without considering `page14.ftl`. This will likely mean some locales end up with mixed language content, where `page10.ftl` is translated, but `page14.ftl` is rendered in English. We could try and use `ftl_has_messages()` to do some conditional rendering, but that might be difficult if there's not a fallback string for everything.

A simpler approach this might be to stick to using just one `page14.ftl` file here, and treat this like a new page. It would mean a handful of duplicate strings between files, but tools like Pontoon support fuzzy string matching to make this less of an issue.

nathan-barrett replied:
TIL that pontoon has fuzzy string matching, ty!

alexgibson reviewed (changes requested):
Page looks good! A couple of small issues with attribution and L10n files, but once those are sorted out this is good to go r+wc

alexgibson commented:
I almost forgot, you also need to add a config path for the new FTL file so that it gets picked up in Pontoon: https://github.com/mozilla/bedrock/blob/main/l10n/configs/pontoon.toml

nathan-barrett commented:
Wanted to note some translations that were provided by marketing that we can use so this would be available for DE and FR ASAP:

|  EN | FR | DE |   
|---|---|---|
|  This Cybersecurity Month, take 20% off Mozilla VPN | 20% de réduction sur Mozilla VPN pour le mois de la cybersécurité  | Erhalte 20 % Rabatt auf Mozilla VPN im Cybersecurity Month  |  
| Use code at checkout for 20% off VPNPRIVACY20 | Entrez le code VPNPRIVACY20 au paiement pour profiter de 20% de réduction  |  Nutze beim Checkout den Code VPNPRIVACY20 für 20 % Rabatt. | 
|  It’s Cybersecurity Month — time to level up your privacy | Es ist Cybersecurity Month – Zeit für ein Upgrade deiner Privatsphäre   |  C’est le mois de la cybersécurité : renforcez votre confidentialité en ligne  |

nathan-barrett commented:
@alexgibson I added a corresponding PR in [www-config](https://github.com/mozmeao/www-config/pull/481) for the switch - and also added the coupon code section

alexgibson commented on bedrock/firefox/templates/firefox/welcome/page14.html:
```diff
@@ -0,0 +1,98 @@
+        {{ vpn_product_referral_link(
+          referral_id='welcome-page14',
+          link_text=ftl('welcome-page14-get-mozilla-vpn'),
+          class_name='mzp-t-product mzp-t-xl',
```
Suggestion (non-blocking): you can add `page_anchor='#pricing',` here if you wanted to link directly to the pricing section on the landing page. Not sure if this is a requirement or not, just calling it out.

alexgibson commented on bedrock/firefox/templates/firefox/welcome/page14.html:
```diff
@@ -0,0 +1,98 @@
+{% endblock %}
+
+{% block js %}
+  {{ js_bundle("firefox_welcome_page14") }}
```
Issue (blocking): this script will throw an error when `SWITCH_VPN_MOMENTS_COUPON_CODE=off`, so suggest wrapping it in a conditional:

```suggestion
  {% if show_coupon %}
    {{ js_bundle("firefox_welcome_page14") }}
  {% endif %}
```

alexgibson commented on bedrock/firefox/templates/firefox/welcome/page14.html:
```diff
@@ -0,0 +1,98 @@
+{% set coupon_code = 'VPNPRIVACY20' %}
+
+{% block shoulder_cta %}{% endblock %}
+
+
```
NIt: remove empty lines

alexgibson commented on l10n/en/firefox/welcome/page14.ftl:
```diff
@@ -0,0 +1,31 @@
+welcome-page14-page-title-alt = It’s Cybersecurity Month — time to level up your privacy
+
+welcome-page14-main-description = { -brand-name-mozilla-vpn } (Virtual Private Network) protects your entire internet connection on your computer, your tablet, and even your phone. For even more security, it comes with a 30-day money back guarantee.
+
```
Nit: remove empty line

alexgibson commented on media/js/firefox/welcome/welcome14.js:
```diff
@@ -0,0 +1,25 @@
+(function () {
+    'use strict';
+
+    var copyButton = document.getElementById('code-copy');
```
Nit: prefer `const` over `var`
```suggestion
    const copyButton = document.getElementById('code-copy');
```

alexgibson reviewed (approved):
A couple of small suggestions/nits and one blocking issue with a JS error, but otherwise this all looks great. r+wc
